File: LocadoraApp/src/app/cadastros/veiculo/model/veiculo.ts
import { Tipo } from "../../tipo/model/tipo";
import { Marca } from "../../marca/model/marca";

export class Veiculo {
    id: number;
    descricao: string;
    placa: string;
    chassi: string;
    idTipo: number;
    idMarca: number;
    ano: number;
    tipo: Tipo;
    marca: Marca;
}

File: LocadoraAPI/LocadoraAPI/Models/Veiculo.cs
using System;
using System.Collections.Generic;

namespace LocadoraAPI.Models
{
    public partial class Veiculo
    {
        public Veiculo()
        {
            Locacao = new HashSet<Locacao>();
        }

        public int Id { get; set; }
        public string Descricao { get; set; }
        public string Placa { get; set; }
        public string Chassi { get; set; }
        public int IdTipo { get; set; }
        public int IdMarca { get; set; }
        public int Ano { get; set; }

        public Marca IdMarcaNavigation { get; set; }
        public Tipo IdTipoNavigation { get; set; }
        public ICollection<Locacao> Locacao { get; set; }
    }
}
